src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,162 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public void unlink(Node<T> node) {
+        if (node == head) {
+            head = node.next;
+            size--;
+        } else if (node == tail) {
+            tail = node.prev;
+            size--;
+        } else {
+            node.next.prev = node.prev;
+            node.prev.next = node.next;
+            size--;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (size == 0) {
+            head = tail = node;
+        } else {
+            node = new Node<>(tail, value, null);
+            tail.next = node;
+            node.prev = tail;
+            node.next = null;
+            tail = node;
+        }
```

Comment:
```suggestion
        Node<T> node = new Node<>(tail, value, null);
        tail = node;
        if (size == 0) {
            head = node;
        } else {
            tail.prev.next = tail;
        }
```
It can be done in a much simpler way